tasks/main.yml

```diff
@@ -1,2 +1,209 @@
 ---
 
+- name: Install postfix
+  package:
+    name:
+      - postfix
+      - postfix-ldap
+      - postfix-pgsql
+    state: present
+
+- name: Setup LDAP authentication
+  block:
+    - name: Install saslauthd
+      package:
+        name:
+          - libsasl2-modules
+          - sasl2-bin
+        state: present
+
+    - name: Place LDAP connection config file
+      template:
+        src: saslauthd.conf.j2
+        dest: /etc/saslauthd.conf
+
+    - name: Place SMTP SASL config file
+      copy:
+        src: smtpd.conf
+        dest: /etc/postfix/sasl/smtpd.conf
+
+    - name: Place saslauthd-postfix config file
+      copy:
+        src: saslauthd-postfix
+        dest: /etc/default/saslauthd-postfix
+
+    - name: Create socket directory for saslauthd
+      shell: dpkg-statoverride --add root sasl 755 /var/spool/postfix/var/run/saslauthd
+      register: dpkg_sasl
+      changed_when: "'already exists' not in dpkg_sasl.stderr"
+      failed_when: "dpkg_sasl.rc != 0 and 'already exists; aborting' not in dpkg_sasl.stderr"
+
+    - name: Add group sasl to user postfix
+      user:
+        name: postfix
+        append: True
+        groups: sasl
+
+  when: postfix_sasl_enabled is defined
+  notify: Restart saslauthd service
+
+- name: Disable and stop saslauthd
+  block:
+    - name: Install python-apt
```

Comment:
Alernative implementations (which are not perfect either):

- First mask the service (this can be done even if it does not exist)
  - Then it can also be stopped and disabled
  - However, then there is this fake service even on machines where this feature was never enabled.
- Instead of  using the list of packages use the list of services (`systemctl list-units`)

Reply:
> This scenario, where the service is installed and running and then needs to be disabled and stopped, will be very rare.

We already ran into this scenario on mail-out01.

Reply:
Fixed it using another way:

```yaml
  register: postfix_service_stop
  failed_when: postfix_service_stop is failed and 'Could not find the requested service ' not in postfix_service_stop.msg
```